### PR TITLE
NETOBSERV-1633 Konflux pipeline multiarch and downstream requirements

### PR DESCRIPTION
**Pull request by OlivierCazade**
## Description

- pipeline refactoring to only have one pipeline definition
- adding a dedicated downstream Dockerfile
- adding multi arch build support

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [X] Will this change affect NetObserv / Network Observability operator? If not, you can ignore the rest of this checklist.
* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

**Comment by OlivierCazade** [comment]
/hold

PR not ready but need to run konflux to test

File: contrib/docker/Dockerfile.downstream [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
+COPY pkg/ pkg/
+
+RUN git status --porcelain
+RUN GOARCH=$TARGETARCH make build_code
+
+# final stage
+FROM --platform=linux/$TARGETARCH registry.access.redhat.com/ubi9/ubi-minimal:9.4
+
+COPY --from=builder /app/flowlogs-pipeline /app/
+COPY --from=builder /app/confgenerator /app/
```
Comment (jotak) [inline contrib/docker/Dockerfile.downstream:25]: iirc in cpaas midstream we did not include the confgenerator, it was only used in the upstream image

File: contrib/docker/Dockerfile.downstream [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
+
+# expose ports
+EXPOSE 2055
```
Comment (jotak) [inline contrib/docker/Dockerfile.downstream:28]: i'd also remove the expose in the downstream, that's a legacy thing for ipfix/netflows

File: contrib/docker/Dockerfile.downstream [hunk 3]
```diff
@@ -1,22 +1,40 @@
-#!/usr/bin/env bash
+ARG TARGETARCH
+FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:v1.22.5-202407301806.g4c8b32d.el9 as builder
```
Comment (jotak) [inline contrib/docker/Dockerfile.downstream:2]: is that going to be automatically tracked and updated by konflux bot?
currently in midstream we're on go 1.22.7

File: contrib/docker/Dockerfile.downstream [hunk 4]
```diff
@@ -1,22 +1,40 @@
...
-cat <<EOF >>"${CONTAINER_FILE}"
 LABEL com.redhat.component="network-observability-flowlogs-pipeline-container"
 LABEL name="network-observability-flowlogs-pipeline"
 LABEL io.k8s.display-name="Network Observability Flow-Logs Pipeline"
 LABEL io.k8s.description="Network Observability Flow-Logs Pipeline"
 LABEL summary="Network Observability Flow-Logs Pipeline"
 LABEL maintainer="support@redhat.com"
 LABEL io.openshift.tags="network-observability-flowlogs-pipeline"
-LABEL upstream-vcs-ref="${COMMIT}"
 LABEL upstream-vcs-type="git"
```
Comment (jpinsonneau) [inline contrib/docker/Dockerfile.downstream:39]:
```suggestion
```

if you remove upstream-vcs-ref we should remove both

File: .tekton/pipeline-ref.yaml [hunk 5]
```diff
@@ -0,0 +1,435 @@
...
+    type: string
+  - default: []
+    description: Array of --build-arg values ("arg=value" strings) for buildah
+    name: build-args
+    type: array
+  - default: ""
+    description: Path to a file with build arguments for buildah, see https://www.mankier.com/1/buildah-build#--build-arg-file
+    name: build-args-file
+    type: string
+  - default: ["linux/x86_64", "linux/arm64"]
```
Comment (jotak) [inline .tekton/pipeline-ref.yaml:77]: ibm power/z are not available?

File: contrib/docker/Dockerfile.downstream (hunk 3 above)
Comment (OlivierCazade) [reply contrib/docker/Dockerfile.downstream:2]: Yes konflux will automatically upgrade it.

File: .tekton/pipeline-ref.yaml (hunk 5 above)
Comment (OlivierCazade) [reply .tekton/pipeline-ref.yaml]: Fixed

**Review by jotak — APPROVED** [review]
LGTM
with 2 small suggestions to make it more similar to what we have in cpaas, but no blocker

**Pull request merged**